File: lc-blind75/ProductOfArrayExceptSelf(lc-238)/Bruteforce/ProductSelf.py
def productExceptSelf(nums):
    n=len(nums)
    res=[]
    for i in range(n):
        pro=1
        for j in range(n):
            if i==j:
                continue
            pro*=nums[j]
        res.append(pro)
    return res

n=int(input())
nums=list(map(int,input().split()))
res=productExceptSelf(nums)
print(res)

File: lc-blind75/ProductOfArrayExceptSelf(lc-238)/Bruteforce/ProductSelf.java
//tc:O(n^2) sc:O()
import java.util.*;
public class ProductSelf {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        int nums[]=new int [n];
        for(int i=0;i<n;i++){
            nums[i]=sc.nextInt();
        }
        int res[]=produceExceptSelf(nums);
        for(int i=0;i<n;i++){
            System.out.println(res[i]);
        }
        sc.close();
    }
    public static int[] produceExceptSelf(int nums[]){
        int n=nums.length;
        int prod[]=new int[n];
        for(int i=0;i<n;i++){
            int pro=1;
            for(int j=0;j<n;j++){
                if(i==j){
                    continue;
                }
                pro*=nums[j];
            }
            prod[i]=pro;
        }
        return prod;
    }
}
